Newspack/Newspack/Stores/AssetStore.swift

```diff
@@ -395,4 +541,315 @@ extension AssetStore {
+        } catch {
+            let error = error as NSError
+            LogError(message: error.localizedDescription)
+        }
+        return assets
+    }
+
+    /// Get the StoryAsset from the specified StoryFolders that has the specified remoteID.
+    ///
+    /// - Parameters:
+    ///   - folders: The StoryFolders, one of which owns the StoryAsset.
+    ///   - remoteIDs: The remoteID of the StoryAsset.
+    /// - Returns: A StoryAsset instance or nil.
+    ///
+    func getStoryAsset(for folders: [StoryFolder], with remoteID: Int64) -> StoryAsset? {
+        guard let context = folders.first?.managedObjectContext else {
+            return nil
+        }
+
+        let fetchRequest = StoryAsset.defaultFetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "folder IN %@ AND remoteID == %d", folders, remoteID)
+        do {
+            return try context.fetch(fetchRequest).first
+        } catch {
+            let error = error as NSError
+            LogError(message: error.localizedDescription)
+        }
+        return nil
+    }
+
+    /// Get a list of StoryAssets for the specified StoryFolder that have changes
+    /// needing to be uploaded.
+    ///
+    /// - Parameter storyFolder: The StoryFolder that owns the StoryAssets
+    /// - Returns: An array of StoryAssets
+    ///
+    func getStoryAssetsWithChanges(storyFolders: [StoryFolder]) -> [StoryAsset] {
+        let context = CoreDataManager.shared.mainContext
+        let fetchRequest = StoryAsset.defaultFetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "folder IN %@ AND modified > synced", storyFolders)
+        if let results = try? context.fetch(fetchRequest) {
+            return results
+        }
+
+        return [StoryAsset]()
+    }
+
+    /// Get a list of StoryAssets needing to have their respective media file uploaded.
+    ///
+    /// - Parameter storyFolder: The StoryFolder that owns the StoryAssets.
+    /// - Returns: An array of StoryAssets.
+    ///
+    func getStoryAssetsNeedingUpload(storyFolder: StoryFolder) -> [StoryAsset] {
+        let context = CoreDataManager.shared.mainContext
+        let fetchRequest = StoryAsset.defaultFetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "folder == %@ AND remoteID == 0 AND type != 'text'", storyFolder)
+        if let results = try? context.fetch(fetchRequest) {
+            return results
+        }
+
+        return [StoryAsset]()
+    }
+
+}
+
+// MARK: - Sync related
+
+extension AssetStore {
+
+    /// Syncs remote assets for the current site's stories.
+    ///
+    /// - Parameter onComplete: A block called after syncing is complete,
+    /// or if there was an error.
+    ///
+    func syncRemoteAssets(onComplete: @escaping (Error?) -> Void) {
+        let folderStore = StoreContainer.shared.folderStore
+        let folders = folderStore.getStoryFoldersWithPosts()
+        let folderUUIDs = folders.map { $0.uuid! }
```

Comment:
Suggestion!! (To get rid of the IUO!)

```suggestion
        let folderUUIDs = folders.compactMap { $0.uuid }
```